rust/src/datatypes.rs

```diff
@@ -310,3 +310,15 @@ impl From<&Dictionary> for pb::Dictionary {
         }
     }
 }
+
+/// Returns true if Lance supports writing this datatype with nulls.
+pub(crate) fn lance_supports_nulls(datatype: &DataType) -> bool {
+    match datatype {
+        DataType::Utf8
+        | DataType::LargeUtf8
+        | DataType::Binary
+        | DataType::FixedSizeBinary(_)
+        | DataType::FixedSizeList(_, _) => true,
+        _ => false,
+    }
+}
```

Comment:
Are there types I am missing?

Reply:
`struct`?

Reply:
Ah yes `list`. `struct` I am not sure. How do we encode structs? They don't have offsets so I don't think we support nulls there.

Reply:
hmm . good point, we think we don't really encode struct itself, but for each field, it encodes separately. 

Let's get it going for now.